src/test/java/com/cloudbees/jenkins/GitHubCommitNotifierTest.java

```diff
@@ -96,7 +97,7 @@ public void testNoBuildRevision() throws Exception {
         FreeStyleProject prj = jRule.createFreeStyleProject();
         prj.setScm(new GitSCM("http://non.existent.git.repo.nowhere/repo.git"));
         prj.getPublishersList().add(new GitHubCommitNotifier());
-        Build b = prj.scheduleBuild2(0).get();
+        Build b = prj.scheduleBuild2(0, new Cause.UserIdCause(), new BuildData()).get();
```

Comment:
On 2.x when the `testNoBuidRevision` executes there is no `BuildData` attached to the run,   `build.getAction(BuildData.class); ` returns null on `ObjectId#getCommitSHA1` so the end result is exactly the same as when `testNoBuildData` is executed 

So instead of trying to mock GitHub or change any version I thought the best curse of action is to make sure there is always a non null `BuildData` on that test